plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/model/CodeModernizerArtifact.kt

```diff
@@ -71,18 +74,21 @@ open class CodeModernizerArtifact(
             return TransformationSummary(summaryFile.readText())
         }
 
+        private fun getSummaryFile(manifest: CodeModernizerManifest) = tempDir.toPath().resolve(manifest.summaryRoot).resolve(summaryNameInZip).toFile()
```

Comment:
if we're essentially only using the tempdir as java NIO, why isn't it just defined as `val tempDir = Paths.get(createTempDir)`

Reply:
also not sure why we're sharing a single tempdir for the lifetime of the artifact, instead of having a new one per artifact

Reply:
We don't always do `tempDir.toPath()` (ex. https://github.com/aws/aws-toolkit-jetbrains/blob/55039796b88f7fa5c1d254ec1a0db6f71c3cd59f/plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/model/CodeModernizerArtifact.kt#L78)

so will leave this as-is for now.